app/src/Repository/MsgrcptSearchRepository.php

```diff
@@ -0,0 +1,203 @@
+use App\Entity\MessageStatus;
+use App\Amavis\DeliveryStatus;
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\ORM\Query;
+use Doctrine\ORM\Tools\Pagination\Paginator;
+
+class MsgrcptSearchRepository extends MsgrcptRepository
+{
+    /**
+     * @param ?array{sort: string, direction: string} $sortParams
+     */
+    public function getSearchQuery(
+        User $user,
+        ?MessageStatus $messageStatus = null,
+        ?string $searchKey = null,
+        ?array $sortParams = null,
+        ?int $fromDate = null
+    ): Query {
+
+
+        $qb = $this->getSearchQueryBuilder($user, $messageStatus);
+        $this->addSearchKeyCondition($qb, $searchKey);
+
+        if (!is_null($fromDate)) {
+            $qb->andWhere('m.timeNum > ' . $fromDate);
+        }
+
+        if ($sortParams) {
+            $qb->orderBy($sortParams['sort'], $sortParams['direction']);
+        }
+
+        $query = $qb->getQuery();
+
+        return $query;
+    }
+
+    public function countByType(?User $user = null, ?int $type = null): int
+    {
+        $messageStatus = null;
+        if ($type) {
+            $entityManager = $this->getEntityManager();
+            $messageStatus = $entityManager->getRepository(MessageStatus::class)->find($type);
+        }
+        $query = $this->getSearchQuery($user, $messageStatus);
+        $paginator = new Paginator($query);
+
+        return $paginator->count();
+    }
+
+    /**
+     * @return array<int, array<string, mixed>>
+     */
+    public function countByTypeAndDays(?User $user = null, ?int $type = null): array
+    {
+        $messageStatus = null;
+        if ($type) {
+            $entityManager = $this->getEntityManager();
+            $messageStatus = $entityManager->getRepository(MessageStatus::class)->find($type);
+        }
+        $qb = $this->getSearchQueryBuilder($user, $messageStatus);
+        $qb->select('COUNT(mr.mailId) as nb_result, m.timeIso, SUBSTRING(m.timeIso, 1, 8) as date_group');
+        $qb->groupBy('date_group');
+
+        $query = $qb->getQuery();
+
+        return $query->getScalarResult();
+    }
+
+    private function getSearchQueryBuilder(
+        User $user,
+        ?MessageStatus $messageStatus = null
+    ): QueryBuilder {
+
+        $qb = $this->createQueryBuilder('mr');
+        $qb->select('mr')
+        ->leftJoin('App\Entity\Msgs', 'm', Join::WITH, 'm.mailId = mr.mailId AND m.partitionTag = mr.partitionTag')
+        ->leftJoin('App\Entity\Maddr', 'maddr', Join::WITH, 'maddr.id = mr.rid');
+
+        $this->addUserSpecificJoins($qb, $user);
+
+        if ($user->isAdmin()) {
+            $this->addDomainCondition($qb, $user);
+        } else {
+            $this->addRecipientsCondition($qb, $user);
+        }
+
+        if (!$messageStatus) {
+            $this->addSpamCondition($qb, $user, false);
+        } elseif ($messageStatus->getId() === MessageStatus::SPAMMED) {
+            $this->addSpamCondition($qb, $user, true);
+        }
```

Comment:
Juste pour être sûr de ce qu’on fait ici : 

- si le statut est `null` (donc `UNTREATED`), on veut être sûr de ne pas afficher les mails qui sont considérés comme du spam
- si le statut est `SPAMMED`, on ne veut lister que les mails considérés comme du spam

En prenant en compte les changements liés à la gestion de `MessageStatus`, je crois que ceci serait équivalent :

```suggestion
        if ($messageStatus === MessageStatus::UNTREATED) {
            $this->addSpamCondition($qb, $user, isSpam: false);
        } elseif ($messageStatus === MessageStatus::SPAMMED) {
            $this->addSpamCondition($qb, $user, isSpam: true);
        }
```

Au passage, j’aime bien expliciter le nom de l’argument quand je passe une variable "brute" (ici `true` ou `false`) parce que ça aide à comprendre à quoi sert cet argument.